gtk/steam-imcontext-gtk.c

```diff
@@ -129,3 +125,34 @@ static void steam_im_context_focus_in(GtkIMContext *context)
         g_warning("Error opening Steam keyboard: %s", error->message);
     }
 }
+
+static gboolean steam_im_context_filter_keypress(GtkIMContext *context, GdkEventKey *event)
+{
+    SteamIMContext *im_context = STEAM_IM_CONTEXT(context);
+
+    if (event->type == GDK_KEY_RELEASE) {
+        if (event->keyval == GDK_KEY_BackSpace) {
+	    gboolean tmp;
+            g_signal_emit_by_name(im_context, "delete_surrounding", -1, 0, &tmp);
+	    return TRUE;
+        } else if (event->keyval == GDK_KEY_Delete) {
+	    gboolean tmp;
+            g_signal_emit_by_name(im_context, "delete_surrounding", 0, 1, &tmp);
+	    return TRUE;
+	} else if (event->keyval == GDK_KEY_Left || event->keyval == GDK_KEY_Right || event->keyval == GDK_KEY_Up || event->keyval == GDK_KEY_Down) {
+            // TODO: Find a way to implement these
+	    return TRUE;
+	}
+
+	gchar keyval_utf8[7];
+	gint length = g_unichar_to_utf8(gdk_keyval_to_unicode(event->keyval), keyval_utf8);
+	keyval_utf8[length] = '\0';
+
+        g_signal_emit_by_name(im_context, "commit", keyval_utf8);
+
+	return TRUE;
```

Comment:
Need to check for a valid/non-control Unicode character before committing it:

```suggestion
        gunichar ch = gdk_keyval_to_unicode(event->keyval);
        if (ch != 0 && !g_unichar_iscntl(ch)) {
            gchar keyval_utf8[8] = { 0, };
	    g_unichar_to_utf8(ch, keyval_utf8);
	    g_signal_emit_by_name(im_context, "commit", keyval_utf8);
	    return TRUE;
	}
```